src/test_util/mod.rs

```diff
@@ -701,3 +726,588 @@ where
     sender.await.expect("Failed to send data");
     events
 }
+
+pub fn sink(channel_size: usize) -> (impl Stream<Item = EventArray>, MockSinkConfig) {
+    let (tx, rx) = SourceSender::new_with_buffer(channel_size);
+    let sink = MockSinkConfig::new(tx, true);
+    (rx.into_stream(), sink)
+}
+
+pub fn sink_with_data(
+    channel_size: usize,
+    data: &str,
+) -> (impl Stream<Item = EventArray>, MockSinkConfig) {
+    let (tx, rx) = SourceSender::new_with_buffer(channel_size);
+    let sink = MockSinkConfig::new_with_data(tx, true, data);
+    (rx.into_stream(), sink)
+}
+
+pub fn sink_failing_healthcheck(
+    channel_size: usize,
+) -> (impl Stream<Item = EventArray>, MockSinkConfig) {
+    let (tx, rx) = SourceSender::new_with_buffer(channel_size);
+    let sink = MockSinkConfig::new(tx, false);
+    (rx.into_stream(), sink)
+}
+
+pub fn source() -> (SourceSender, MockSourceConfig) {
+    let (tx, rx) = SourceSender::new_with_buffer(1);
+    let source = MockSourceConfig::new(rx);
+    (tx, source)
+}
+
+pub fn source_with_data(data: &str) -> (SourceSender, MockSourceConfig) {
+    let (tx, rx) = SourceSender::new_with_buffer(1);
+    let source = MockSourceConfig::new_with_data(rx, data);
+    (tx, source)
+}
+
+pub fn source_with_event_counter() -> (SourceSender, MockSourceConfig, Arc<AtomicUsize>) {
+    let event_counter = Arc::new(AtomicUsize::new(0));
+    let (tx, rx) = SourceSender::new_with_buffer(1);
+    let source = MockSourceConfig::new_with_event_counter(rx, Arc::clone(&event_counter));
+    (tx, source, event_counter)
+}
+
+pub fn transform(suffix: &str, increase: f64) -> MockTransformConfig {
+    MockTransformConfig::new(suffix.to_owned(), increase)
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(default)]
+pub struct MockSourceConfig {
+    #[serde(skip)]
+    receiver: Arc<Mutex<Option<LimitedReceiver<EventArray>>>>,
```

Comment:
Would it be worth moving these types into a submodule, say, `test_util::mock`? This file is already pretty long.

Reply:
Yeah, that's a good idea.

Ironically, every time I look at this file, I also think "wow, this is really long and hard to find what I want"... and then apparently do nothing to be the change I want to see in the world. :)

Reply:
Addressed in d14568bc3.

Broke things apart in a fairly granular manner, but this is in anticipation of doing more consolidation and building out of specialized components needed for tests, so that we can continue to try and reduce the necessary feature flags to run common unit tests.